volcano 1.12 scheduler panic caused by pod scheduled by volcano 1.7 and using volcano.sh/gpu-number resources

### Description

In the upgrade scenario from volcano1.7+volcano-device-plugin to volcano1.12+hami-device-plugin (with VGPUEnable),pods scheduled by volcano 1.7 and using the volcano.sh/gpu-number resource will cause the scheduler pod in volcano 1.12 to panic in func ni.Others[gpushare.DeviceName].(Devices).AddResource(pod) ,where gs.Device is nil.


### Steps to reproduce the issue

1.Deploy Volcano 1.7 and the volcano-device-plugin, and then start a Pod that requests the volcano.sh/gpu-number resource.
2.Upgrade the Volcano version to v1.12+hami-device-plugin, Configure Volcano to use the VGPUEnable configuration option. 

### Describe the results you received and expected

**result receive:** 
volcano v1.12 scheduler panic with following messages: 
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x1d24d68]

goroutine 483 [running]:
k8s.io/apimachinery/pkg/util/runtime.handleCrash({0x2dcdff8, 0x469c060}, {0x25c1940, 0x46364f0}, {0x469c060, 0x0, 0x10000000043a900?})
	/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:89 +0xee
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc000d00e00?})
	/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:59 +0x108
panic({0x25c1940?, 0x46364f0?})
	/usr/local/go/src/runtime/panic.go:791 +0x132
volcano.sh/volcano/pkg/scheduler/api/devices/nvidia/gpushare.(*GPUDevices).AddResource(0x0, 0xc000a73688)
	/go/src/volcano.sh/volcano/pkg/scheduler/api/devices/nvidia/gpushare/device_info.go:106 +0x88
volcano.sh/volcano/pkg/scheduler/api.(*NodeInfo).addResource(0xc000552480, 0xc000a73688)
	/go/src/volcano.sh/volcano/pkg/scheduler/api/node_info.go:498 +0xa3
volcano.sh/volcano/pkg/scheduler/api.(*NodeInfo).setNode(0xc000552480, 0xc000742c08)
	/go/src/volcano.sh/volcano/pkg/scheduler/api/node_info.go:386 +0x9aa
volcano.sh/volcano/pkg/scheduler/api.(*NodeInfo).SetNode(0xc0007103c0, 0xc000742c08)
	/go/src/volcano.sh/volcano/pkg/scheduler/api/node_info.go:330 +0x4c
volcano.sh/volcano/pkg/scheduler/cache.(*SchedulerCache).AddOrUpdateNode(0xc000742008, 0xc000742c08)
	/go/src/volcano.sh/volcano/pkg/scheduler/cache/event_handlers.go:498 +0x105
volcano.sh/volcano/pkg/scheduler/cache.(*SchedulerCache).SyncNode(0xc000742008, {0xc0001b4120, 0x15})
	/go/src/volcano.sh/volcano/pkg/scheduler/cache/event_handlers.go:624 +0x4cc
volcano.sh/volcano/pkg/scheduler/cache.(*SchedulerCache).processSyncNode(0xc000742008)
	/go/src/volcano.sh/volcano/pkg/scheduler/cache/cache.go:1152 +0x18b
volcano.sh/volcano/pkg/scheduler/cache.(*SchedulerCache).runNodeWorker(...)
	/go/src/volcano.sh/volcano/pkg/scheduler/cache/cache.go:1140
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x30?)
	/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:226 +0x33
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000b12dd0, {0x2da1940, 0xc000813920}, 0x1, 0xc00011a770)
	/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:227 +0xaf
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000b12dd0, 0x0, 0x0, 0x1, 0xc00011a770)
	/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:204 +0x7f
k8s.io/apimachinery/pkg/util/wait.Until(...)
	/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:161
created by volcano.sh/volcano/pkg/scheduler/cache.(*SchedulerCache).Run in goroutine 1
	/go/src/volcano.sh/volcano/pkg/scheduler/cache/cache.go:764 +0x8f

**result expect:**
scheduler running normally

### What version of Volcano are you using?

1.12.1

### Any other relevant information

_No response_

--- a/pkg/scheduler/api/node_info.go
+++ b/pkg/scheduler/api/node_info.go
@@ -495,13 +495,22 @@ func (ni *NodeInfo) RemoveTask(ti *TaskInfo) error {
 
 // addResource is used to add sharable devices
 func (ni *NodeInfo) addResource(pod *v1.Pod) {
-	ni.Others[gpushare.DeviceName].(Devices).AddResource(pod)
+	// In the upgrade scenario from volcano1.7+volcano-device-plugin to volcano1.12+hami-device-plugin (with VGPUEnable),
+	// pods scheduled by volcano 1.7 and using the volcano.sh/gpu-number resource will cause the scheduler pod in volcano 1.12 to panic
+	// at dev := gs.Device[id] ,where gs.Device is nil.
+
+	// Add an if judgment condition to fix the panic.
+	if gpushare.GpuSharingEnable || gpushare.GpuNumberEnable {
+		ni.Others[gpushare.DeviceName].(Devices).AddResource(pod)
+	}
 	ni.Others[vgpu.DeviceName].(Devices).AddResource(pod)
 }
 
 // subResource is used to subtract sharable devices
 func (ni *NodeInfo) subResource(pod *v1.Pod) {
-	ni.Others[gpushare.DeviceName].(Devices).SubResource(pod)
+	if gpushare.GpuSharingEnable || gpushare.GpuNumberEnable {
+		ni.Others[gpushare.DeviceName].(Devices).SubResource(pod)
+	}
 	ni.Others[vgpu.DeviceName].(Devices).SubResource(pod)
 }
 

